pkg/velero/server/args.go

```diff
@@ -0,0 +1,159 @@
+	args := []string{"server"}
+	// we are overriding args, so recreate args from scratch
+	if len(dpaFeatureFlags) > 0 {
+		args = append(args, fmt.Sprintf("--features=%s", strings.Join(dpaFeatureFlags, ",")))
+	}
+	if boolptr.IsSetToTrue(a.DefaultVolumesToFsBackup) {
+		args = append(args, "--default-volumes-to-fs-backup=true")
+	} else if boolptr.IsSetToFalse(a.DefaultVolumesToFsBackup) {
+		args = append(args, "--default-volumes-to-fs-backup=false")
+	}
+	if logLevel != "" {
+		logrusLevel, err := logrus.ParseLevel(logLevel)
+		if err != nil {
+			return []string{}, fmt.Errorf("invalid log level %s, use: %s", logLevel, "trace, debug, info, warning, error, fatal, or panic")
+		}
+		args = append(args, fmt.Sprintf("--log-level=%s", logrusLevel.String()))
+	}
+	if a.BackupSyncPeriod != nil {
+		args = append(args, fmt.Sprintf("--backup-sync-period=%s", a.BackupSyncPeriod.String())) // duration
+	}
+	if a.ClientBurst != nil {
+		args = append(args, fmt.Sprintf("--client-burst=%s", strconv.Itoa(*a.ClientBurst))) // int
+	}
+	if a.ClientPageSize != nil {
+		args = append(args, fmt.Sprintf("--client-page-size=%s", strconv.Itoa(*a.ClientPageSize))) // int
+	}
+	if a.ClientQPS != nil {
+		if _, err := strconv.ParseFloat(*a.ClientQPS, 32); err != nil {
+			return nil, err
+		}
+		args = append(args, fmt.Sprintf("--client-qps=%s", *a.ClientQPS)) // float32
+	}
+	// default-backup-storage-location set outside Args
+	if a.DefaultBackupTTL != nil {
+		args = append(args, fmt.Sprintf("--default-backup-ttl=%s", a.DefaultBackupTTL.String())) // duration
+	}
+	if a.DefaultItemOperationTimeout != nil {
+		args = append(args, fmt.Sprintf("--default-item-operation-timeout=%s", a.DefaultItemOperationTimeout.String())) // duration
+	}
+	if a.ResourceTimeout != nil {
+		args = append(args, fmt.Sprintf("--resource-timeout=%s", a.ResourceTimeout.String())) // duration
+	}
+	if a.RepoMaintenanceFrequency != nil {
+		args = append(args, fmt.Sprintf("--default-repo-maintain-frequency=%s", a.RepoMaintenanceFrequency.String())) // duration
+	}
+	// default-volume-snapshot-locations set outside Args
+	if a.DisabledControllers != nil {
+		args = append(args, fmt.Sprintf("--disable-controllers=%s", strings.Join(a.DisabledControllers, ","))) // strings
+	}
+	if a.GarbageCollectionFrequency != nil {
+		args = append(args, fmt.Sprintf("--garbage-collection-frequency=%s", a.GarbageCollectionFrequency.String())) // duration
+	}
+	if a.FormatFlag != "" {
+		args = append(args, fmt.Sprintf("--log-format=%s", a.FormatFlag)) // format
+	}
+	if a.MetricsAddress != "" {
+		args = append(args, fmt.Sprintf("--metrics-address=%s", a.MetricsAddress)) // string
+	}
+	// plugin-dir is fixed to /plugins
+	if a.ProfilerAddress != "" {
+		args = append(args, fmt.Sprintf("--profiler-address=%s", a.ProfilerAddress)) // string
+	}
+	if a.PodVolumeOperationTimeout != nil {
+		args = append(args, fmt.Sprintf("--fs-backup-timeout=%s", a.PodVolumeOperationTimeout.String())) // duration
```

Comment:
could this cause a conflict with https://github.com/openshift/oadp-operator/blob/master/controllers/velero.go#L568 ?

Reply:
Yes. This is intentional.. when one can specify in two places, the server args wins.

Reply:
we're making new args array here so what's there is not used.